src/delaunator.cpp

```diff
@@ -229,76 +227,73 @@ Delaunator::Delaunator(vector<double> in_coords) {
-    const sort_to_center s = {
-        .cx = m_center_x,
-        .cy = m_center_y,
-        .coords = coords
-    };
-    sort(ids.begin(), ids.end(), s);
-    // quicksort(ids, coords, 0, n - 1, m_center_x, m_center_y);
-    // cerr << ids << endl;
+    
+    std::sort(ids.begin(), ids.end(), [&](int64_t i, int64_t j)
+    {
+        return compare(coords, i, j, m_center_x, m_center_y) < 0;
+    });
 
     m_hash_size = ceil(sqrt(n));
     m_hash.reserve(m_hash_size);
     for (int i = 0; i < m_hash_size; i++) m_hash.push_back(-1);
 
-    m_hull.reserve(coords.size());
+    m_hl.reserve(coords.size());
 
-    long int e = insert_node(i0);
+    m_hull_index = insert_node(i0,coords);
+    int64_t e = m_hull_index;
     hash_edge(e);
-    m_hull[e].t = 0;
+    m_hl[e].t = 0;
 
-    e = insert_node(i1, e);
+    e = insert_node(i1, e,coords);
     hash_edge(e);
-    m_hull[e].t = 1;
+    m_hl[e].t = 1;
 
-    e = insert_node(i2, e);
+    e = insert_node(i2, e,coords);
     hash_edge(e);
-    m_hull[e].t = 2;
+    m_hl[e].t = 2;
 
-    const long int max_triangles = 2 * n - 5;
+    const int64_t max_triangles = 2 * n - 5;
     triangles.reserve(max_triangles * 3);
     halfedges.reserve(max_triangles * 3);
     add_triangle(i0, i1, i2, -1, -1, -1);
+    
     double xp = NAN;
     double yp = NAN;
-    for (long int k = 0; k < n; k++) {
-        const long int i = ids[k];
+   
+    for (int64_t k = 0; k < ids.size(); k++)
+    {
+        const int64_t i = ids[k];
         const double x = coords[2 * i];
         const double y = coords[2 * i + 1];
-        if (x == xp && y == yp) continue;
+
+        if (k > 0 && std::abs(x - xp) <= m_epilon && std::abs(y - yp) <= m_epilon) continue;
+        
         xp = x;
         yp = y;
+        
         if (
                 (x == i0x && y == i0y) ||
                 (x == i1x && y == i1y) ||
                 (x == i2x && y == i2y)
         ) continue;
 
-        const long int start_key = hash_key(x, y);
-        long int key = start_key;
-        long int start = -1;
+        const int64_t start_key = hash_key(x, y);
+        int64_t key = start_key;
+        int64_t start = -1;
         do {
             start = m_hash[key];
             key = (key + 1) % m_hash_size;
         } while(
-            (start < 0 || m_hull[start].removed) &&
+            (start < 0 || m_hl[start].removed) &&
             (key != start_key)
         );
+        
+        start   = m_hl[start].prev;
+        e       = start;
 
-        e = start;
-
-        while(
-            area(
-                x, y,
-                m_hull[e].x, m_hull[e].y,
-                m_hull[m_hull[e].next].x, m_hull[m_hull[e].next].y
-            ) >= 0
-        ) {
-            e = m_hull[e].next;
+        while(!orient(x, y,m_hl[e].x, m_hl[e].y,m_hl[m_hl[e].next].x, m_hl[m_hl[e].next].y))
+        {
+            e = m_hl[e].next;
 
             if (e == start) {
                 throw runtime_error("Something is wrong with the input points.");
```

Comment:
This place is different in JS — let's port over the fix here too.

Reply:
very good, i missed that. fixed now